app/main.py

```diff
@@ -1,25 +1,96 @@
-from abc import ABC
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return instance.__dict__[self.name]
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"{self.name} must "
+                             f"be between {self.min_amount} "
+                             f"and {self.max_amount}.")
+        instance.__dict__[self.name] = value
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self.name = name
```

Comment:
Consider making 'name' a protected attribute  for better encapsulation like this self.name = f"_{name}"